### PR TITLE
Consume ENV vars where possible

### PR DESCRIPTION
**Pull request by mkolesnik**
Instead of consuming flags, consume environment variables where
possible.
Some flags are retained for backwards compatibility not to break
consumers, and will be removed once their consumers are updated.

This is the first step in resolving #887 

Depends on #886 

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
       - name: Make sure each commit in the PR is within reviewable size
         uses: ./gh-actions/commit-size
         with:
-          size: 200
+          size: 250
```
Comment (skitt) [inline .github/workflows/linting.yml:43]: If this gets updated for any reason, could you mention this change in the corresponding commit message?

File: Makefile.inc [hunk 2]
```diff
@@ -31,73 +31,58 @@ include $(SHIPYARD_DIR)/Makefile.versions
 # Process extra flags from the `using=a,b,c` optional flag
 
 ifneq (,$(filter libreswan,$(_using)))
-override DEPLOY_ARGS += --cable_driver libreswan
+export CABLE_DRIVER = libreswan
```
Comment (skitt) [inline Makefile.inc]: Since these are exported further down, how about

    CABLE_DRIVER = lighthouse

Same for the other variables.

**Review by skitt — CHANGES_REQUESTED** [review]

File: scripts/shared/lib/utils [hunk 3]
```diff
@@ -189,3 +189,10 @@ function load_settings_from_yaml() {
         cluster_subm["${cluster}"]=$(_yq ".submariner")
     done
 }
+
+function print_env() {
+   echo "Running with environment variables:"
+   for var; do
+       eval echo "${var}=\${${var}@Q}"
+   done
```
Comment (skitt) [inline scripts/shared/lib/utils]: The safe way of doing this is

```
    for var; do
        declare -n value=$var
        printf '%s=%q\n' "$var" "$value"
    done
```

**Review by skitt — CHANGES_REQUESTED** [review]
Regarding the cache-removing commit, note that we only use `buildx` for multi-arch images.

File: scripts/shared/lib/utils (hunk 3 above)
Comment (mkolesnik) [reply scripts/shared/lib/utils]: Cool, good to know :D

**Comment by mkolesnik** [comment]
> Regarding the cache-removing commit, note that we only use `buildx` for multi-arch images.

Yes you're right of course, it's BuildKit that has the improvements (well probably buildx too as BuildKit is it's backed).
I updated the commit message.

**Review by skitt — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**